src/image_intel.cpp

```diff
@@ -41,7 +41,8 @@ static size_t getBootAddress()
  */
 static void setBootAddress(size_t address)
 {
-    std::ignore = exec(FW_SETENV_CMD " bootcmd bootm %08x", address);
+    std::ignore =
+        exec(FW_SETENV_CMD " bootcmd bootm %08x 2>/dev/null", address);
```

Comment:
What if `fw_setenv` truly fails? We've been ignoring the return code but we could at least see the error message. Now as we hide it, I guess we need to add some error checking.

Reply:
No,  we've been ignoring `stdout` data in this case.
This `exec` is not the same as the Linux syscall. It defined in the [`src/subprocess.cpp`](https://github.com/YADRO-KNS/obmc-yadro-fwupdate/blob/df232c6aff807d75c7564135806cebfb0fe8db56/src/subprocess.cpp#L37).
If `fw_setenv` will truly fail we get an exception that will be handled in the `main()`.

Reply:
cool. that's why it's not usually a good idea to name your methods identical to POSIX or other standard calls.

Reply:
By the way, the Linux syscall with the exact same name does not exist in C/C++. There are [several with suffixes](https://man7.org/linux/man-pages/man3/exec.3.html).